scripts/gui/microbe_hud.mjs

```diff
@@ -384,33 +428,87 @@ function onExitToMenuClicked() {
     }
 }
 
+
 //! Updates the mouse hover box with stuff
 function updateHoverInfo(vars){
 
     const panel = document.getElementById("mouseHoverPanel");
     common.clearChildren(panel);
 
-    panel.appendChild(document.createTextNode("Stuff at " + vars.mousePos + ":"));
+    if(showMouseCoordinates) {
+        panel.appendChild(document.createTextNode("Stuff at " + vars.mousePos + ":"));
+        panel.appendChild(document.createElement("br"));
+    }
 
-    if(vars.noCompounds){
+    const mainContent = document.createElement("div");
+    mainContent.style.width = "100%";
+    mainContent.style.height = "15px";
+    mainContent.style.color = "white";
+    mainContent.style.textAlign = "center";
+    mainContent.innerHTML = "Hello";
 
-        panel.appendChild(document.createElement("br"));
-        panel.appendChild(document.createTextNode("Nothing to eat here"));
+    if(vars.noCompounds){
 
+        mainContent.innerHTML = "Nothing to eat here";
+        panel.appendChild(mainContent);
     } else {
 
-        common.getKeys(vars).forEach(function(key){
+        if(!vars.compounds){
+            mainContent.innerHTML = "Error reading compounds Data";
+            return;
+        }
+        let objCompoundsData = null;
+
+
 
-            // Skip things that are handled elsewhere
-            if(key == "mousePos" || key == "hoveredCells")
+        if(typeof vars.compounds === "string" || vars.compounds instanceof String){
+            try{
+                objCompoundsData = JSON.parse(vars.compounds);
+            } catch(err){
+                mainContent.innerHTML = "invalid json for mouseHover info: " + err;
                 return;
+            }
+        } else {
+            objCompoundsData = vars.compounds;
+        }
+
+        // Compounds data are store in json format, so we need parse it
+        objCompoundsData = JSON.parse(vars.compounds);
```

Comment:
This is a duplicate call to JSON.parse? there is one with proper error catching a few lines above this.